src/main/scala/com/fulcrumgenomics/bam/ErrorRateByReadPosition.scala

```diff
@@ -0,0 +1,280 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2016 Fulcrum Genomics LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+package com.fulcrumgenomics.bam
+
+import com.fulcrumgenomics.cmdline.{ClpGroups, FgBioTool}
+import com.fulcrumgenomics.util.{Metric, NumericCounter, ProgressLogger}
+import dagr.commons.CommonsDef._
+import dagr.commons.io.{Io, PathUtil}
+import dagr.commons.util.LazyLogging
+import dagr.sopt._
+import htsjdk.samtools.SAMFileHeader.SortOrder
+import htsjdk.samtools.reference.ReferenceSequenceFileFactory
+import htsjdk.samtools.util.SequenceUtil
+import htsjdk.samtools.util.SequenceUtil.{isValidBase => validBase}
+import htsjdk.samtools.{CigarOperator, SAMRecord, SamReaderFactory}
+
+@clp(description =
+  """
+    |Calculates the error rate by read position on mapped BAMs.
+    |
+    |If a reference is given, the sort order must be coordinate.  If no reference is given, the MD tag must be present
+    |for every mapped read and computed correctly, and no sort order is required.
+    |
+    |The output metrics text file will have the extension 'error_rate_by_read_position.txt'.  The output file will
+    |contain a row per read and per position in that read.  The 'read_number' will be 0 for fragments, 1 for the first
+    |end of a pair, and 2 for the second end of a pair. Both the error rate and number of bases observed will be
+    |provided.  Soft-clipped bases, and unmapped, secondary, supplementary, and vendor failing reads are ignored.
+    |Optionally, duplicates are ignored.
+  """,
```

Comment:
Would be nice to describe here what constitutes an error.  Specifically how are the following handled:
1. No-calls
2. Insertions
3. Deletions